File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    ans = 0
    # 끝점을 기준으로 정렬
    targets.sort(key=lambda x: x[1])
    last_e = 0
    for s,e in targets:
        if s >= last_e:
            ans += 1
            last_e = e
    return ans

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.kt
class Solution {
    fun solution(targets: Array<IntArray>): Int {
        var ans = 0
        targets.sortBy { it[1] }
        var l_e = -300001
        for ((s,e) in targets) {
            if (s >= l_e) {
                ans++
                l_e = e
            }
        }
        return ans
    }
}